api/src/main/scala/za/co/absa/loginsvc/rest/service/AuthSearchService.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.rest.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import za.co.absa.loginsvc.model.User
+import za.co.absa.loginsvc.rest.config.auth.{ActiveDirectoryLDAPConfig, UsersConfig}
+import za.co.absa.loginsvc.rest.config.provider.AuthConfigProvider
+import za.co.absa.loginsvc.rest.config.validation.ConfigValidationException
+
+import java.util
+import javax.naming.Context
+import javax.naming.directory.{Attributes, DirContext, SearchControls, SearchResult}
+import javax.naming.ldap.{Control, InitialLdapContext, PagedResultsControl}
+import scala.collection.JavaConverters.enumerationAsScalaIteratorConverter
+
+@Service
+class AuthSearchService @Autowired()(authConfigProvider: AuthConfigProvider) {
+
+  private val logger = LoggerFactory.getLogger(classOf[AuthSearchService])
+
+  private val usersConfig = authConfigProvider.getUsersConfig
+  private val adLDAPConfig = authConfigProvider.getLdapConfig
+
+  private val configs = Array(usersConfig, adLDAPConfig).filter(_.order != 0).sortBy(_.order)
+  if (configs.isEmpty)
+    throw ConfigValidationException("No authentication method enabled in config")
+
+  def searchUser(username: String): User = {
+
+    configs.foreach {
+      case u: UsersConfig =>
+        val result = searchUsersConfig(username)
+        if (result.isDefined) return result.get
+      case l: ActiveDirectoryLDAPConfig =>
+        val result = searchLDAP(username)
+        if (result.isDefined) return result.get
+      case _ => //ignore for now
+    }
+    throw new NoSuchElementException(s"Value not found in any object.")
+  }
+
+  private def searchUsersConfig(username: String): Option[User] = {
+    usersConfig.knownUsersMap.get(username).flatMap { userConfig =>
+      val optionalAttributes: Map[String, Option[AnyRef]] = userConfig.attributes.getOrElse(Map.empty).map {
+        case (k, v) => (k, Some(v))
+      }
+      Some(User(username, userConfig.groups.toList, optionalAttributes))
+    }
+  }
+
+  private def searchLDAP(username: String): Option[User] = {
```

Comment:
I suggest that these two implementations (searching in built-in users/in LDAP) exist in two separate classes that both implement the same trait, e.g.
```scala
trait UserSearchable {
  def searchForUser: Option[User] 
}
```